standard/navbox.lua

```diff
@@ -1,447 +1,500 @@
-	local ret = 2
-	if args.imageleft then ret = ret + 1 end
-	if args.image then ret = ret + 1 end
-	return ret
+	return titleCell
 end
 
-local function renderAboveRow(tbl)
-	if not args.above then return end
-
-	tbl:tag('tr')
-		:tag('td')
-			:addClass('navbox-abovebelow')
-  			:addClass('wiki-backgroundcolor-light')
-			:addClass(args.aboveclass)
-			:cssText(args.basestyle)
-			:cssText(args.abovestyle)
-			:attr('colspan', getAboveBelowColspan())
-			:tag('div')
-				:wikitext(processItem(args.above, args.nowrapitems))
+function NavBox._checkAgainstNavboxTemplate()
+	return mw.getCurrentFrame():getParent():getTitle():gsub('/sandbox$', '') ~= 'Template:Navbox'
 end
 
-local function renderBelowRow(tbl)
-	if not args.below then return end
-
-	tbl:tag('tr')
-		:tag('td')
-			:addClass('navbox-abovebelow')
-  			:addClass('wiki-backgroundcolor-light')
-			:addClass(args.belowclass)
-			:cssText(args.basestyle)
-			:cssText(args.belowstyle)
-			:attr('colspan', getAboveBelowColspan())
-			:tag('div')
-				:wikitext(processItem(args.below, args.nowrapitems))
-end
-
---
---   List rows
---
-local function renderListRow(tbl, index, listnum)
-	local row = tbl:tag('tr')
+function NavBox._aboveBelowRow(body, rowType)
+	if String.isEmpty(_args[rowType]) then
+		return body
+	end
 
-	if index == 1 and args.imageleft then
-		row
+	body:tag('tr')
 			:tag('td')
-				:addClass('navbox-image')
-				:addClass(args.imageclass)
-				:css('width', '1px')               -- Minimize width
-				:css('padding', '0px 2px 0px 0px')
-				:cssText(args.imageleftstyle)
-				:attr('rowspan', #listnums)
+				:addClass('navbox-abovebelow')
+				:addClass('wiki-backgroundcolor-light')
+				:addClass(_args[rowType .. 'class'])
+				:cssText(_args.basestyle)
+				:cssText(_args[rowType .. 'style'])
+				:attr('colspan', NavBox._getAboveBelowColspan())
 				:tag('div')
-					:wikitext(processItem(args.imageleft))
+					:wikitext(NavBox._processItem(_args[rowType], _args.nowrapitems))
+
+	return body
+end
+
+function NavBox._getAboveBelowColspan()
+	local colSpan = 2
+	if _args.imageleft then
+		colSpan = colSpan + 1
+	end
+	if _args.image then
+		colSpan = colSpan + 1
 	end
+	return colSpan
+end
 
-	if args['group' .. listnum] then
-		local groupCell = row:tag('th')
+function NavBox._processItem(item, noWrapItems)
+	if item:sub(1, 2) == '{|' then
+		return '\n' .. item ..'\n'
+	end
 
-		groupCell
+	if noWrapItems == 'yes' then
+		local lines = {}
+		for line in (item .. '\n'):gmatch('([^\n]*)\n') do
+			local prefix, content = line:match('^([*:;#]+)%s*(.*)')
+			if prefix and not content:match('^<span class="nowrap">') then
+				line = prefix .. '<span class="nowrap">' .. content .. '</span>'
+			end
+			table.insert(lines, line)
+		end
+		item = table.concat(lines, '\n')
+	end
+
+	if item:match('^[*:;#]') then
+		return '\n' .. item ..'\n'
+	end
+
+	return item
+end
+
+local _ODD_EVEN_MARKER = '\127_ODDEVEN_\127'
+local _RESTART_MARKER = '\127_ODDEVEN0_\127'
```

Comment:
Can't StringUtils.interpolate be used instead?

Reply:
feel free to try it out^^
https://liquipedia.net/starcraft2/Module:Hjpalpha/sandbox7

Reply:
https://liquipedia.net/starcraft2/index.php?title=Module%3AHjpalpha%2Fsandbox7&type=revision&diff=2116121&oldid=2115358

Please test that it works as expected, I don't know all the details of how the navbox is supposed to work.

Reply:
played around a bot with it, so far it looks good, will test some more later today